src/thunderbots/software/param_server/cfg/Params.cfg

```diff
@@ -50,6 +50,12 @@ navigator_params = gen.add_group("Passing")
 navigator_params.add("static_field_position_quality_x_offset", double_t, 0, "", 0.3, 0, 1)
 navigator_params.add("static_field_position_quality_y_offset", double_t, 0, "", 0.3, 0, 1)
 navigator_params.add("static_field_position_quality_friendly_goal_distance_weight", double_t, 0, "", 0.3, 0, 1)
+navigator_params.add("num_passes_to_optimize", int_t, 0, "", 50, 0, 1000)
+navigator_params.add("num_passes_to_keep_after_pruning", int_t, 0, "", 10, 0, 1000)
+navigator_params.add("number_of_gradient_descent_steps_per_iter", int_t, 0, "", 20, 0, 1000)
+navigator_params.add("pass_equality_max_position_difference_meters", double_t, 0, "", 0.1, 0, 4)
+navigator_params.add("pass_equality_max_start_time_difference_seconds", double_t, 0, "", 0.5, 0, 3)
+navigator_params.add("pass_equality_max_speed_difference_meters_per_second", double_t, 0, "", 0.3, 0, 4)
```

Comment:
Should have asked before, but are these dynamic parameters thread safe?

Reply:
@akhilveeraghanta - can you confirm?

Reply:
I think we should be good, but only because I only read from them.

Reply:
Just my 2c, but we should be fine. @garethellis0 is correct we only ever read the parameters in the AI (AI never sets them), and ROS is responsible for safely updating them when they change externally.

Reply:
Sounds good

Reply:
The ROS Param server api itself is _not_ thread safe, but we don't directly set the parameters managed by the DynamicReconfigure through `ros::param::set`, we do it through the UI which makes a service call to the reconfigure server, which is thread safe. So we should be fine as long as we use the gui, if we start setting them directly we can run into problems.